qutebrowser/browser/commands.py

```diff
@@ -526,6 +526,35 @@ def tab_detach(self):
         cur_widget = self._current_widget()
         self._tabbed_browser.close_tab(cur_widget, add_undo=False)
 
+    @cmdutils.register(instance='command-dispatcher', scope='window')
+    def tab_attach(self, win_id=None):
+        """Attach the current tab to a different window.
+
+        Args:
+            win_id: The id of the window to attach to or None for
+                    the first-opened window
+        """
+        window = None
+        if win_id is None:
+            window = objreg.window_by_index(0)
+        else:
+            if int(win_id) not in objreg.window_registry:
+                raise cmdexc.CommandError(
+                    "There's no window with id {}!".format(win_id))
+            window = objreg.window_registry[int(win_id)]
+
+        if window.win_id == self._win_id:
+            raise cmdexc.CommandError(
+                "Tab is already attached to that window!")
+
+        url = self._current_url()
+        cur_widget = self._current_widget()
+        window.tabbed_browser.tabopen(url, explicit=True)
+        if self._count() < 2 and window.win_id != self._win_id:
+            objreg.window_registry[self._win_id].close()
```

Comment:
I was thinking that if a user has tabs.last-close set to "ignore", they would still want `:tab-attach` to delete their current window if they are attaching its only tab to another window. But thinking about it more, a user might want to send a tab to many different windows or something, so I guess I should just use `close_tab` here, you are correct.